Userland/Libraries/LibTextCodec/ShiftJISConverter.h

```diff
@@ -0,0 +1,7064 @@
+ * Copyright (c) 2021, Łukasz Patron <priv.luk@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Types.h>
+
+namespace {
+// https://unicode.org/Public/MAPPINGS/OBSOLETE/EASTASIA/JIS/SHIFTJIS.TXT
+u32 convert_shift_jis_to_utf8(u16 in)
+{
+    switch (in) {
+
+#define MAP(X, Y) \
+    case X:       \
+        return Y
+
+        MAP(0x20, 0x0020);
+        MAP(0x21, 0x0021);
+        MAP(0x22, 0x0022);
+        MAP(0x23, 0x0023);
+        MAP(0x24, 0x0024);
+        MAP(0x25, 0x0025);
+        MAP(0x26, 0x0026);
+        MAP(0x27, 0x0027);
+        MAP(0x28, 0x0028);
+        MAP(0x29, 0x0029);
+        MAP(0x2A, 0x002A);
+        MAP(0x2B, 0x002B);
+        MAP(0x2C, 0x002C);
+        MAP(0x2D, 0x002D);
+        MAP(0x2E, 0x002E);
+        MAP(0x2F, 0x002F);
+        MAP(0x30, 0x0030);
+        MAP(0x31, 0x0031);
+        MAP(0x32, 0x0032);
+        MAP(0x33, 0x0033);
+        MAP(0x34, 0x0034);
+        MAP(0x35, 0x0035);
+        MAP(0x36, 0x0036);
+        MAP(0x37, 0x0037);
+        MAP(0x38, 0x0038);
+        MAP(0x39, 0x0039);
+        MAP(0x3A, 0x003A);
+        MAP(0x3B, 0x003B);
+        MAP(0x3C, 0x003C);
+        MAP(0x3D, 0x003D);
+        MAP(0x3E, 0x003E);
+        MAP(0x3F, 0x003F);
+        MAP(0x40, 0x0040);
+        MAP(0x41, 0x0041);
+        MAP(0x42, 0x0042);
+        MAP(0x43, 0x0043);
+        MAP(0x44, 0x0044);
+        MAP(0x45, 0x0045);
+        MAP(0x46, 0x0046);
+        MAP(0x47, 0x0047);
+        MAP(0x48, 0x0048);
+        MAP(0x49, 0x0049);
+        MAP(0x4A, 0x004A);
+        MAP(0x4B, 0x004B);
+        MAP(0x4C, 0x004C);
+        MAP(0x4D, 0x004D);
+        MAP(0x4E, 0x004E);
+        MAP(0x4F, 0x004F);
+        MAP(0x50, 0x0050);
+        MAP(0x51, 0x0051);
+        MAP(0x52, 0x0052);
+        MAP(0x53, 0x0053);
+        MAP(0x54, 0x0054);
+        MAP(0x55, 0x0055);
+        MAP(0x56, 0x0056);
+        MAP(0x57, 0x0057);
+        MAP(0x58, 0x0058);
+        MAP(0x59, 0x0059);
+        MAP(0x5A, 0x005A);
+        MAP(0x5B, 0x005B);
+        MAP(0x5C, 0x00A5);
+        MAP(0x5D, 0x005D);
+        MAP(0x5E, 0x005E);
+        MAP(0x5F, 0x005F);
+        MAP(0x60, 0x0060);
+        MAP(0x61, 0x0061);
+        MAP(0x62, 0x0062);
+        MAP(0x63, 0x0063);
+        MAP(0x64, 0x0064);
+        MAP(0x65, 0x0065);
+        MAP(0x66, 0x0066);
+        MAP(0x67, 0x0067);
+        MAP(0x68, 0x0068);
+        MAP(0x69, 0x0069);
+        MAP(0x6A, 0x006A);
+        MAP(0x6B, 0x006B);
+        MAP(0x6C, 0x006C);
+        MAP(0x6D, 0x006D);
+        MAP(0x6E, 0x006E);
+        MAP(0x6F, 0x006F);
+        MAP(0x70, 0x0070);
+        MAP(0x71, 0x0071);
+        MAP(0x72, 0x0072);
+        MAP(0x73, 0x0073);
+        MAP(0x74, 0x0074);
+        MAP(0x75, 0x0075);
+        MAP(0x76, 0x0076);
+        MAP(0x77, 0x0077);
+        MAP(0x78, 0x0078);
+        MAP(0x79, 0x0079);
+        MAP(0x7A, 0x007A);
+        MAP(0x7B, 0x007B);
+        MAP(0x7C, 0x007C);
+        MAP(0x7D, 0x007D);
```

Comment:
(aside from 0x5C) these are all handled by the default case already, so do not need a switch case

Reply:
Fixed.